core/src/SYCL/Kokkos_SYCL_Parallel_Team.hpp

```diff
@@ -487,9 +488,11 @@ class ParallelFor<FunctorType, Kokkos::TeamPolicy<Properties...>,
 
     // Functor's reduce memory, team scan memory, and team shared memory depend
     // upon team size.
-    auto& space = *m_policy.space().impl_internal_space_instance();
+    auto& space       = *m_policy.space().impl_internal_space_instance();
+    m_scratch_pool_id = space.acquire_team_scratch_space();
```

Comment:
Technically we could wait for the event completion until `ParallelFor::execute()` gets called right?
(Not asking you to change but it might be worth a comment)